desdeo/mcdm/pareto_navigator.py

```diff
@@ -0,0 +1,271 @@
+"""Functions related to the Pareto Navigator method are defined here.
+
+Reference of the method:
+
+Eskelinen, Petri, et al. "Pareto navigator for interactive nonlinear
+multiobjective optimization." OR spectrum 32 (2010): 211-227.
+"""
+
+import numpy as np
+from scipy.optimize import linprog
+from scipy.spatial import ConvexHull
+
+from desdeo.problem import (
+    Problem,
+    get_ideal_dict,
+    get_nadir_dict,
+    numpy_array_to_objective_dict,
+    objective_dict_to_numpy_array
+)
+
+
+def classification_to_reference_point(
+    problem: Problem,
+    pref_info: dict[str, list[str]],
+    current_solution: dict[str, float]
+) -> dict[str, float]:
+    """Convert preference information given as classification into a reference point.
+
+    Args:
+        problem (Problem): The problem being solved.
+        pref_info (dict): The preference information given as classification.
+        current_solution (dict[str, float]): The current solution.
+
+    Returns:
+        dict[str, float]: A reference point converted from classification.
+    """
+    ref = []
+    ideal = objective_dict_to_numpy_array(problem, get_ideal_dict(problem))
+    nadir = objective_dict_to_numpy_array(problem, get_nadir_dict(problem))
+    sol = objective_dict_to_numpy_array(problem, current_solution)
+
+    for i in range(len(pref_info["classification"])):
+        if pref_info["classification"][i] == "<":
+            ref.append(ideal[i])
+        elif pref_info["classification"][i] == ">":
+            ref.append(nadir[i])
+        elif pref_info["classification"][i] == "=":
+            ref.append(sol[i])
+    return numpy_array_to_objective_dict(problem, np.array(ref))
+
+def calculate_adjusted_speed(allowed_speeds: np.ndarray, speed: float) -> float:
+    """Calculate an adjusted speed from a given float.
+
+    Note:
+        Adjusting the speed is not specified in the article but seems necessary.
+
+    Args:
+        allowed_speeds (np.ndarray): An array of allowed speeds.
+        speed (int): A given speed value where.
+
+    Returns:
+        float: An adjusted speed value calculated from given float.
+            Is between 0 and 1.
+    """
+    return (speed / np.max(allowed_speeds)) / 20
```

Comment:
Is the value `20` here necessary to be hard-coded? I suggest it is given as an optional argument here, and its default value can be 20.

Reply:
Muutin tuon jakajan arvon valinnaiseksi argumentiksi funktiolle. Ei sinänsä ole tietoa, kuinka paljon sitä täytyy muuttaa eri ongelmille, kun tuosta ei ollut mainintaa itse artikkelissa.